### PR TITLE
Disable immediate diagnostic reporting

### PR DESCRIPTION
**Pull request by kyouko-taiga**

### CODE REVIEW

File: Sources/Core/Dianostics.swift [hunk 1]
```diff
@@ -1,34 +1,32 @@
...
-  /// is `true`.
-  ///
-  /// - Note: the default is `false` because we build many more tests than command-line tools.
-  public init(reportingToStderr: Bool = false) {
-    self.reportingToStderr = reportingToStderr
+  /// Creates an instance.
+  public init() {}
+
+  /// Creates an instance reporting `d`.
+  public init(_ d: Diagnostic) {
```
Comment (dabrahams) [inline Sources/Core/Dianostics.swift:16]: Why move this code?

**Review by dabrahams — APPROVED** [review]

File: Sources/Core/Dianostics.swift (hunk 1 above)
Comment (kyouko-taiga) [reply Sources/Core/Dianostics.swift:16]: I have this weird obsession about declaring initializers before any other method/computed property. I also like overloaded methods (which initializers are) to be grouped.

These rules are just personal preferences that I'm totally fine with other people not upholding.

Comment (dabrahams) [reply Sources/Core/Dianostics.swift:16]: I understand that, but we should really establish policies so that trivial reformatting doesn't occur.  These kinds of rearrangements have a cost in merge conflicts and ease-of-review.

Comment (kyouko-taiga) [reply Sources/Core/Dianostics.swift:16]: I don't mind starting and maintaining a document with all policies. I'm less motivated to extend swift-format so that it can enforce them automatically.

**Comment by dabrahams** [comment]
If you're going to make a policy about initializer positioning, you need to consider whether to forbid them in extensions, especially when they are protocol requirements.  “Don't make rules that are hard to follow” is a good guideline.
